cmd/subnetcmd/join.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/spf13/cobra"
+)
+
+var avagoConfigPath string
+
+// avalanche subnet deploy
+func newJoinCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "join [subnetName]",
+		Short: "Instruct a validator node to begin validating a new subnet",
+		Long: `Instruct a validator node to begin validating a new subnet.
+The NodeID of that validator node must have been whitelisted by one of the 
+subnet's control keys for this to work.
+
+The node also needs to be restarted.
+If --avalanchego-config is provided, this command tries to edit the config file at that path
+(consider correct permissions).`,
+		RunE: joinCmd,
+		Args: cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&avagoConfigPath, "avalanchego-config", "", "file path of the avalanchego config file")
+	return cmd
+}
+
+func joinCmd(cmd *cobra.Command, args []string) error {
+	/*
+		chains, err := validateSubnetName(args)
+		if err != nil {
+			return err
+		}
+		subnetName := chains[0]
+
+		sc, err := app.LoadSidecar(subnetName)
+		if err != nil {
+			return err
+		}
+
+		subnetID := sc.SubnetID
+		if subnetID == ids.Empty {
+			return errNoSubnetID
+		}
+	*/
+	subnetIDstr := "29Zd5yhP7Yb2cTebBbUVKUjHHNviHzgj1y9kKJsvMn2dyWnkpG"
+	var network models.Network
+	networkStr, err := prompts.CaptureList(
+		"Choose a network to deploy on (this command only supports public networks)",
+		[]string{models.Fuji.String(), models.Mainnet.String()},
+	)
+	if err != nil {
+		return err
+	}
+	network = models.NetworkFromString(networkStr)
+	networkLower := strings.ToLower(network.String())
+
+	if avagoConfigPath == "" {
```

Comment:
Maybe we should make this a prompt instead of just setting via flags.

How would you like update the avalanchego config?
1. Automatic
2. Manual

For manual, we just print instructions to the screen. For automatic, we ask for the path to the avalanchego config

Reply:
done